store/sqlstore/channel_store.go

```diff
@@ -2094,6 +2094,7 @@ func (s SqlChannelStore) UpdateLastViewedAt(channelIds []string, userId string)
 
```

Comment:
Right before this (I can't comment not-in-context line) you have a early return on postgres, that is the reason why postgres is not doing the update at all.